dev-client/jest/utils.tsx

```diff
@@ -25,14 +34,48 @@ const nativeBaseInset = {
   insets: {top: 0, left: 0, right: 0, bottom: 0},
 };
 
-const TestWrapper = ({children}: React.PropsWithChildren) => {
+type WrapperProps = {
+  initialState?: Partial<AppState>;
+  route?: keyof RootStackParamList;
+};
+
+const TestWrapper = ({
+  initialState,
+  route,
+  children,
+}: React.PropsWithChildren<WrapperProps>) => {
+  const store = useRef(createStore(initialState));
   return (
-    <NativeBaseProvider theme={theme} initialWindowMetrics={nativeBaseInset}>
-      <Portal.Host>{children}</Portal.Host>
-    </NativeBaseProvider>
+    <Provider store={store.current}>
```

Comment:
Question: Out of curiosity, did you just manually go up from a Screen and see what provider-like things there were as parent components?

Reply:
a combination of this and running the test and seeing what error messages emerged!